topo/node/cisco/cisco.go

```diff
@@ -434,6 +453,60 @@ func isNode8000eUp(ctx context.Context, req *rest.Request) bool {
 	return podIsUpRegex.Match(buf.Bytes())
 }
 
+// SpawnCLIConn spawns a CLI connection towards a IOSXR using `kubectl exec` terminal and ensures CLI is ready
+// to accept inputs.
+// scrapligo options can be provided to this function for a caller to modify scrapligo platform.
+// For example, mock transport can be set via options
+func (n *Node) SpawnCLIConn() error {
+	opts := []scrapliutil.Option{
+		scrapliopts.WithAuthBypass(),
+		scrapliopts.WithTimeoutOps(scrapliOperationTimeout),
+	}
+	// add options defined in test package
+	opts = append(opts, n.testOpts...)
+	opts = n.PatchCLIConnOpen("kubectl", []string{"xr"}, opts)
+	if n.Proto.Model != ModelXRD {
+		opts = n.PatchCLIConnOpen("kubectl", []string{"telnet", "0", "60000"}, opts)
+	}
+	var err error
+	n.cliConn, err = n.GetCLIConn(scrapliPlatformName, opts)
+	// TODO: add the following pattern in the scrapli/scrapligo/blob/main/assets/platforms/cisco_iosxr.yaml
+	n.cliConn.FailedWhenContains = append(n.cliConn.FailedWhenContains, "ERROR")
+	n.cliConn.FailedWhenContains = append(n.cliConn.FailedWhenContains, "% Failed")
+	n.cliConn.OnClose = endTelnet
```

Comment:
isn't telnet only used for 8000e? if so does this additional telnet end cmd cause any issues with xrd? there is no good way for me to test this because there are no XRD ops that utilize the CLI conn in this PR

Reply:
you are right, the default onClose is enough for xrd and there is not need to end telnet. guarded the code.